pkg/lockfile/match-nuget-csproj.go

```diff
@@ -0,0 +1,79 @@
+package lockfile
+
+import (
+	"errors"
+	"io"
+	"os"
+
+	"github.com/google/osv-scanner/internal/utility/fileposition"
+	"github.com/google/osv-scanner/pkg/models"
+
+	"path/filepath"
+	"strings"
+)
+
+// NugetCsprojMatcher matches the source file of a Nuget lockfile
+// https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#locking-dependencies
+type NugetCsprojMatcher struct{}
+
+func (m NugetCsprojMatcher) GetSourceFile(lockfile DepFile) (DepFile, error) {
```

Comment:
Is the sourcefile not always located in the same folder as the lockfile?
If the answer is yes, the code could be modified to:
```
return lockfile.Open("project.csproj")
```

Reply:
There is no convention for the name of the csproj file, it could be `project.csproj` or `foobar.csproj`